neptune/internal/notebook.py

```diff
@@ -0,0 +1,72 @@
+#
+# Copyright (c) 2020, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+import threading
+import uuid
+from typing import Any
+
+from neptune.internal.backends.neptune_backend import NeptuneBackend
+from neptune_old.utils import is_ipython
+
+_logger = logging.getLogger(__name__)
+
+_checkpoints_lock = threading.Lock()
+_checkpoints = dict()
+
+
+class Notebook:
```

Comment:
This module should be called global_notebook_context or something like that. And it should expose only global function. Notebook class will probably something else in future.